### ISSUE TITLE
Select-String breaks when used with ForEach-Object -Parallel and Start-ThreadJob

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#11242]
@jszabo98 discovered the `Start-ThreadJob` problem in #11092; it affects `ForEach-Object -Parallel` as well

Unlike #11092, it doesn't seem to be related to output formatting data per se, but to the VT escape sequences created to highlight the matching part of a line, given that the problem goes away when you add `-NoEmphasis` - even though the tests use `-Raw` for direct `[string]` output, in which case `-NoEmphasis` should have no effect, given that `-Raw` doesn't perform highlighting.

# Steps to reproduce

```powershell
# Adding -NoEmphasis to Select-String makes the problem go away.
Start-ThreadJob { 'hi' | Select-String hi -Raw } | Receive-Job -Wait -AutoRemoveJob | Should -be hi
'hi' | ForEach-Object -parallel { $_ | Select-String hi -Raw } | Should -Be hi
```

# Expected behavior

Both tests should pass.

# Actual behavior

Both tests emit an error message and fail:

```none
Select-String: Object reference not set to an instance of an object.
...
 ^ Expected 'hi', but got $null.
```

# Environment data

```none
PowerShell Core 7.0.0-preview.6
```

**Comment by iSazonov** [comment]
@SteveL-MSFT @daxian-dbw We fix NRE from time to time. I hope we turn on nullable in all code base like I started in #10693.

**Comment by iSazonov** [comment]
It was fixed in #11105

**Comment by PaulHigin** [comment]
Forgot to close this issue after fixing it.
